lib/archive/unpack.go

```diff
@@ -74,50 +80,50 @@ func Unpack(op trace.Operation, tarStream io.Reader, filter *FilterSpec, root st
 			return err
 		}
 
-		op.Debugf("processing tar header: %s", header.Name)
-
+		op.Debugf("processing tar header: %#v", *header)
+		op.Debugf("using FilterSpec : (%#v)", *filter)
 		// skip excluded elements unless explicitly included
 		if filter.Excludes(op, header.Name) {
 			continue
 		}
 
 		// fix up path
-		// TODO: update with revised filterspec
 		stripped := strings.TrimPrefix(header.Name, filter.StripPath)
 		rebased := filepath.Join(filter.RebasePath, stripped)
 		absPath := filepath.Join(root, rebased)
+		op.Debugf("attempting to write to target (%s)", absPath)
+
+		op.Debugf("writing %d bytes to: %s", header.Size, absPath)
 
 		switch header.Typeflag {
 		case tar.TypeDir:
 			err = os.MkdirAll(absPath, header.FileInfo().Mode())
 			if err != nil {
+				op.Errorf("Failed to create directory%s: %s", absPath, err)
 				return err
 			}
-			continue
 		case tar.TypeSymlink:
-
 			err := os.Symlink(header.Linkname, absPath)
 			if err != nil {
 				op.Errorf("Failed to create symlink %s->%s: %s", absPath, header.Linkname, err)
 				return err
 			}
-
 		case tar.TypeReg:
```

Comment:
@hickeng do you think we should also include tar.TypeRegA in this case of the switch? just in case?